src/js/rtc_peer_connection_factory.js

```diff
@@ -66,9 +67,13 @@ export default class RtcPeerConnectionFactory {
             clearTimeout(self._idleRtcPeerConnectionTimerId);
             self._idleRtcPeerConnectionTimerId = null;
         }
-        setTimeout(() => {
-            self._requestPeerConnection();
-        }, 0);
+        if (!self._peerConnectionRequestInFlight) {
+            self._peerConnectionRequestInFlight = true;
+            setTimeout(() => {
+                self._requestPeerConnection();
+                self._peerConnectionRequestInFlight = false;
```

Comment:
The `_requestPeerConnection()` is async, right? Shouldn't the `self._peerConnectionRequestInFlight = false;` be inside the `_requestPeerConnection()` method?

Maybe the if statement and `self._peerConnectionRequestInFlight = true;` can also be moved to inside the method.

Reply:
The code inside setTimeout() block is executed asynchronosly. As self._peerConnectionRequestInFlight = false; is within setTimeout() block, it is as good as the statement is inside  _requestPeerConnection() method. If we move the statement  self._peerConnectionRequestInFlight = true; inside the method, we run into the same issue of creating orphaned peer connections which causes leak.

Reply:
Updated PR and moved self._peerConnectionRequestInFlight = true; moved inside the _requestPeerConnection() method.